cli/command.go

```diff
@@ -274,19 +274,49 @@ func (c *BaseCommand) Hidden() bool {
 
 // Prompt prompts the user for a value. If stdin is a tty, it prompts. Otherwise
 // it reads from the reader.
-func (c *BaseCommand) Prompt(msg string) (string, error) {
-	scanner := bufio.NewScanner(io.LimitReader(c.Stdin(), 64*1_000))
+func (c *BaseCommand) Prompt(ctx context.Context, msg string, args ...any) (string, error) {
+	resultCh := make(chan string, 1)
+	errCh := make(chan error, 1)
+
+	// Prompts for user input and returns the value or an error if there is one.
+	// If the context is canceled, this function leaves the c.Stdin in a bad state.
```

Comment:
I think this sentence (`If the context is canceled, this function leaves the c.Stdin in a bad state.`) should go in the godoc, in the comment preceding this function declaration.

Reply:
Moved.